deep_agent/aegra/mcp_auth.py

```
@@ -136,6 +145,70 @@ async def _resolve_sso(self) -> str:
+        cached = self._cc_cache.get(mcp_name)
+        if cached:
+            token, expires_at = cached
+            if (expires_at - time.time()) > _TOKEN_EXPIRY_BUFFER_SECONDS:
+                return token
+
+        oauth_cfg = server_cfg.get("oauth") or {}
+        token_endpoint = oauth_cfg.get("token_endpoint")
+        if not token_endpoint:
+            raise RuntimeError(
+                f"MCP '{mcp_name}': token_endpoint missing for client_credentials grant"
+            )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)